java-support/src/main/java/io/cloudstate/javasupport/controller/ControllerContext.java

```diff
@@ -0,0 +1,19 @@
+package io.cloudstate.javasupport.controller;
+
+import io.cloudstate.javasupport.Metadata;
+import io.cloudstate.javasupport.MetadataContext;
+
+/** Context for controller calls. */
+public interface ControllerContext extends MetadataContext {
+  /**
+   * Get the metadata associated with this call.
+   *
+   * <p>Note, this only returns call level associated metadata. For unary in calls, this will be the
+   * same as the message metadata, but for streamed calls, it will contain metadata associated with
+   * the whole stream, so for example if this was a gRPC call, it will contain the HTTP headers for
+   * that gRPC call.
+   *
+   * @return The call level metadata.
+   */
+  Metadata metadata();
```

Comment:
is it needed here because it is already defined in MetadataContext?

Reply:
It's defined here to override the docs, since for controllers, there are two levels of metadata, and this is documented here.